subsys/bluetooth/mesh/shell/shell_onoff_cli.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <bluetooth/mesh/models.h>
+#include <shell/shell.h>
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "shell_utils.h"
+
+static struct bt_mesh_model *mod;
+
+static int shell_onoff_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod) {
+		if (!shell_model_first_get(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, &mod)) {
+			return -ENODEV;
+		}
+	}
+
+	struct bt_mesh_onoff_cli *cli = mod->user_data;
+	struct bt_mesh_onoff_status rsp;
+
+	int err = bt_mesh_onoff_cli_get(cli, NULL, &rsp);
```

Comment:
I guess this can be added as an additional feature at some point. It is a valid utility that will make the module even more flexible. However I don't think it is worth while reworking the entire addressing pattern at this point.